gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java

```diff
@@ -223,19 +224,20 @@ public static String createSuffixForJobTrigger(LeaseAttemptStatus.LeasedToAnothe
    * the event to revisit. It will update the jobKey to the reminderKey provides and the Properties map to
    * contain the cron scheduler for the reminder event and information about the event to revisit
    * @param originalKey
-   * @param reminderKey
    * @param status
    * @return
    * @throws SchedulerException
    */
-  protected JobDetailImpl createJobDetailForReminderEvent(JobKey originalKey, JobKey reminderKey,
-      LeaseAttemptStatus.LeasedToAnotherStatus status)
+  protected JobDetailImpl createJobDetailForReminderEvent(JobKey originalKey, LeaseAttemptStatus.LeasedToAnotherStatus status)
       throws SchedulerException {
-    JobDetailImpl jobDetail = (JobDetailImpl) this.schedulerService.getScheduler().getJobDetail(originalKey);
-    jobDetail.setKey(reminderKey);
-    JobDataMap jobDataMap = jobDetail.getJobDataMap();
-    jobDataMap = updatePropsInJobDataMap(jobDataMap, status, schedulerMaxBackoffMillis);
-    jobDetail.setJobDataMap(jobDataMap);
+    /* Cloning the original jobDetail creates a new object reference but the jobDataMap and Properties fields are
+    shallow copies that need to be replaced by deep ones to avoid shared references between the original and reminder
+    jobs in the scheduler
+     */
+    JobDetailImpl jobDetail = (JobDetailImpl) this.schedulerService.getScheduler().getJobDetail(originalKey).clone();
+    JobDataMap originalJobDataMap = jobDetail.getJobDataMap();
```

Comment:
because we are cloning in line 237,  `jobDetail.getJobDataMap()` should be named `clonedJobDataMap`

Reply:
I am calling it original because `jobDetail.getJobDataMap` will reference the original jobDataMap which needs to be replaced. The cloning happens in line 261